### PR TITLE
Ecf sync lesson data

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: db/migrate/20160124021330_add_uuid_to_content_access_events.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddUuidToContentAccessEvents < ActiveRecord::Migration
+  def change
+    add_column :content_access_events, :uuid, :string
+    ContentAccessEvent.reset_column_information
+    ContentAccessEvent.all.each { |h| h.update(uuid: SecureRandom.uuid) }
```
Comment (eschlange) [inline db/migrate/20160124021330_add_uuid_to_content_access_events.rb]: I usually end up making these table changes over a series of migrations. I like this series of changes though, it keeps them confined to a single location.

File: app/models/session_event.rb [hunk 2]
```diff
@@ -9,6 +9,10 @@ class SessionEvent < ActiveRecord::Base
 
   scope :accesses, -> { where(event_type: TYPES.access) }
 
+  attr_accessor :lesson_guid
+
+  before_validation :set_uuid, :set_lesson
```
Comment (eschlange) [inline app/models/session_event.rb]: Is there any case where a bad lesson ID value may be passed to a session event?

Comment (ericcf) [inline app/models/session_event.rb]: If the client passes an unrecognized `lesson_guid`, that could happen and an exception would be reported. It shouldn't be possible though so in this instance I'd like to see the exception report.

**Comment by eschlange** [comment]
Looks good!

**Pull request merged**